homeassistant/components/panasonic_viera/__init__.py

```diff
@@ -59,11 +100,146 @@ async def async_setup_entry(hass, config_entry):
 
 async def async_unload_entry(hass, config_entry):
     """Unload a config entry."""
-    return all(
+    unload_ok = all(
         await asyncio.gather(
             *[
                 hass.config_entries.async_forward_entry_unload(config_entry, component)
                 for component in PLATFORMS
             ]
         )
     )
+
+    if unload_ok:
+        hass.data[DOMAIN].pop(config_entry.entry_id)
+
```

Comment:
Is it okay to leave the unload method this way?

Reply:
Yeah, looks good! There are no other integration wide listeners that we should unsubscribe, right?

Reply:
Nope, the panasonic_viera library has no listener services